routemodel/data_retrieval/get_elevations_point_compression.py

```diff
@@ -0,0 +1,165 @@
+                long = new_long
+
+                # Step 4
+                dy *= 2
+                dx *= 2
+
+                # Step 5
+                if dx < 0:
+                    dx = abs(dx) - 1
+                if dy < 0:
+                    dy = abs(dy) - 1
+
+                # Step 6
+                index = int(((dx + dy) * (dx + dy + 1) / 2) + dx)
+
+                # Step 7
+                rem = []
+                while index > 0:
+                    rem.append(int(index % 32))
+                    index = int(index / 32)
+
+                # Step 8
+                for i in range(0, len(rem) - 1):
+                    rem[i] += 32
+
+                # Step 9
+                for i in rem:
+                    compressed_points += encoder_dict[i]
+
+        return compressed_points
+
+    def get_elevation_data(self, points):
+        """
+        Getting elevation data response from Bing Maps API.
+        @param points: Compressed base 64 string containing coordinates data
+        @return: Requests.response object from API call
+        """
+        # append specified elevation URL details
+        url = BASE_URL + 'Elevation/List?'
+
+        # add parameters to URL
+        url += 'points={}&heights={}&key={}'.format(points, self.heights, API_KEY)
+
+        # request URL and return response
+        response = requests.get(url)
+        return response.json()
+
+    def parse_elevation_data(self, response):
+        """
+        Parsing through Elevations API call response.
+        @param response: Requests.response object from API call
+        @return: boolean
+        """
+        # convert response into dict
+        elevations = response
+        # parse down to relevant information
+
+        print("Status Code: {}".format(elevations["statusCode"]))
+
+        try:
+            elevations = elevations['resourceSets'][0]['resources'][0]['elevations']
+        except KeyError:
+            return False
+
+        # create DataFrame with headers
+        headers = ['Latitude', 'Longitude', 'Elevation']
+        elevation_df = pd.DataFrame(columns=headers)
+
+        # loop through waypoints
+        # to write coordinates into DataFrame
+        counter = 0
+        indexCheck = 0
+
+        for index, waypoint in enumerate(self.waypts):
+            for key in waypoint.keys():
+
+                try:
+                    indexCheck = elevations[counter]
+                except:
+                    break
+
+                elevation_df = elevation_df.append({'Latitude': key,
+                                                    'Longitude': self.waypts[-1][key],
+                                                    'Elevation': elevations[counter]},
+                                                   ignore_index=True)
+                counter += 1
+
+        # write dataframe into csv and return true
+        elevation_df.to_csv(ELEVATIONS_FILE)
+        print(f'Elevations written to CSV file.')
+        return True

```

Comment:
A lot of this is repetitive from get_elevations.py. Why did you choose to create a close copy rather than inheriting?

Reply:
Again I apologize for this the inherited copy would require me to make some changes to original get_elevations and the changes I would need to make broke some tests and I would need to redesign the tests. This just ensured I got resulting elevations we needed.